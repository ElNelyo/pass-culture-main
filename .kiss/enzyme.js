// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
