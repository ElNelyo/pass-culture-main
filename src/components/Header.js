import React from 'react'
import { connect } from 'react-redux'

import Hamburger from './Hamburger'
import { THUMBS_URL } from '../utils/config'

const Header = ({ user, venue }) => {
  return (
    <div className='header flex items-center justify-start p2'>
      <Hamburger className='hamburger mr1'/>
      <div className='header__logo'>Pass Culture</div>
      <div className='flex-auto' />
      {
        user && venue && (
          <div className='header__venue flex items-center'>
            <span className='mx1'>
              {venue.name}
            </span>
            <img className='header__venue__image mr2'
              alt='venue'
              src={`${THUMBS_URL}/venues/${venue.id}`} />
          </div>
        )
      }
      {
        user && (
          <img className='header__avatar'
            alt='avatar'
            src={`${THUMBS_URL}/users/${user.id}`} />
        )
      }
      {
        user && !venue && (
          <div className='header__account-balance'>
            {user.account}{user.account ? 'â‚¬' : ''}
          </div>
        )
      }
    </div>
  )
}

export default connect(
  ({ user }) => ({
    user,
    venue: user && user.userVenues && user.userVenues[0] && user.userVenues[0].venue
  })
)(Header)
