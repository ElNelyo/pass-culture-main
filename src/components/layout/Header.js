import classnames from 'classnames'
import get from 'lodash.get'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'

import Icon from './Icon'
import SignoutButton from './SignoutButton'
import Logo from './Logo'

class Header extends Component {
  constructor() {
    super()
    this.state = {
      showMobileMenu: false,
    }
  }

  render() {
    const {
      name,
      whiteHeader
    } = this.props
    const {
      showMobileMenu
    } = this.state
    return (
      <header className={classnames(
        'navbar',
        { 'is-primary': !whiteHeader }
      )}>
      <div className="container">
        <div className="navbar-brand">
          <Logo className="navbar-item" whiteHeader={whiteHeader} />
          <span className="navbar-burger" onClick={e => this.setState({
            showMobileMenu: !showMobileMenu
          })}>
          <span></span>
          <span></span>
          <span></span>
        </span>
      </div>
      <div className={classnames("navbar-menu", {
        'is-active': showMobileMenu
      })}>
      <div className="navbar-end">
        {
          !whiteHeader && [
            <NavLink className="navbar-item" to={'/guichet'} key={0}>
            <span className='icon'><Icon svg={'ico-guichet-w'} /></span>
            { false && [
              <span>Guichet</span>
            ]}
          </NavLink>,
            <NavLink className="navbar-item" to={'/offres'} key={1}>
            <span className='icon'><Icon svg={'ico-offres-w'} /></span>
            <span>Vos offres</span>
          </NavLink>
        ]
    }
      <div className="navbar-item has-dropdown is-hoverable">
        <NavLink className="navbar-link" to="#">
          <span className='icon'>
            {!whiteHeader ?
              <Icon svg='ico-user-circled-w' />
              : <Icon svg='ico-user-circled' />
            }
          </span>
          <span>
            {name}
          </span>
        </NavLink>
        <div className="navbar-dropdown is-right">
          <NavLink to={'/profil'} className='navbar-item'>
          <span className='icon'><Icon svg={'ico-user'} /></span>
          <span>Profil</span>
        </NavLink>
        <NavLink to={'/structures'} className='navbar-item'>
          <span className='icon'><Icon svg={'ico-structure'} /></span>
          <span>Structures</span>
        </NavLink>
        { false && [
          <NavLink to={'/delegations'} className='navbar-item'>
            <span className='icon'><Icon svg={'ico-delegation'} /></span>
            <span>Délégations</span>
          </NavLink>
          ,
          <NavLink to={'/comptabilite'} className='navbar-item'>
            <span className='icon'><Icon svg={'ico-compta'} /></span>
            <span>Comptabilité</span>
          </NavLink>
        ]}
        <SignoutButton tagName='a' className='navbar-item'>
          <span className='icon'><Icon svg={'ico-deconnect'} /></span>
          <span>Déconnexion</span>
        </SignoutButton>
    </div>
  </div>
</div>
</div>
</div>
</header>
)
}
}

export default connect(
  state => ({
    name: get(state, 'user.publicName')
  })
)(Header)
