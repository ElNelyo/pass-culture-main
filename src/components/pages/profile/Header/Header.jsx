import PropTypes from 'prop-types'
import React from 'react'

import { LOCALE_FRANCE, MONTH_IN_NUMBER, YEAR_IN_NUMBER } from '../../../../utils/date/date'
import { NON_BREAKING_SPACE } from '../../../../utils/specialCharacters'
import User from '../ValueObjects/User'

const Header = ({ user }) => {
  const {
    publicName,
    deposit_expiration_date: depositExpirationDateIso,
    wallet_balance: walletBalance,
  } = user

  const formattedExpirationDate = new Date(depositExpirationDateIso).toLocaleDateString(
    LOCALE_FRANCE,
    {
      ...YEAR_IN_NUMBER,
      ...MONTH_IN_NUMBER,
      day: 'numeric',
    }
  )

  return (
    <section className="ph-wrapper">
      <div className="ph-pseudo">
        {`${publicName}`}
      </div>
      <div className="ph-wallet-balance">
        {`${walletBalance}${NON_BREAKING_SPACE}€`}
      </div>
      <div className="ph-end-validity-date">
        {`crédit valable jusqu’au ${formattedExpirationDate}`}
      </div>
    </section>
  )
}

Header.propTypes = {
  user: PropTypes.shape(User).isRequired,
}

export default Header
