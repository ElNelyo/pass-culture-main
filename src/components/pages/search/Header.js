import React from 'react'

const renderPageHeader = searchPageTitle => (
  <header className="no-dotted-border">
    <h1 className="is-normal fs19">
      {searchPageTitle}
    </h1>
  </header>
)

export default renderPageHeader
