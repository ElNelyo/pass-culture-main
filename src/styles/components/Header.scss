.header {
  height: 55px;
  background-color: black;
  color: white;
  position: fixed;
  top: 0;
  width: 100%;
  .hamburger {
    margin-left: -10px;
  }
  &__logo {
    position: absolute;
    left:50%;
    transform: translate(-50%, 0);
    text-align: center;
    font-size: 20px;
  }
  &__account-balance {
    position: absolute;
    right: 45px;
    padding-top: 3px;
  }
  &__avatar {
    height: 40px;
    position: absolute;
    right: 0;
  }
}
