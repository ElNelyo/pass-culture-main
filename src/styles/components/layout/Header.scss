header.navbar {

  .navbar-brand {
    .navbar-item {
      font-size: $size-medium;
      // font-weight: $weight-light;
      // font-style: italic;

      // img {
      //   margin: -0.5rem;
      //   margin-right: 0;
      //   max-height: 2.75rem;
      // }

      // sub {
      //   font-size: 0.4em;
      //   text-align: top;
      //   position: relative;
      //   top: -0.5em;
      //   left: 0.3em;
      //   font-style: normal;
      //   font-weight: $weight-semi-bold;
      // }
    }
  }

}