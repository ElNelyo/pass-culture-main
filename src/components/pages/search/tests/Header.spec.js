import React from 'react'
import { shallow } from 'enzyme'

import Header from '../Header'

describe('src | components | pages | search | Header', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      // when
      const wrapper = shallow(<Header />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
