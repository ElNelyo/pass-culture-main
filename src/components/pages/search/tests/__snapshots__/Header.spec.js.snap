// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | search | Header snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <renderPageHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="is-normal fs19"
      >
        Object {}
      </h1>,
      "className": "no-dotted-border",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Object {},
        "className": "is-normal fs19",
      },
      "ref": null,
      "rendered": Object {},
      "type": "h1",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="is-normal fs19"
        >
          Object {}
        </h1>,
        "className": "no-dotted-border",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Object {},
          "className": "is-normal fs19",
        },
        "ref": null,
        "rendered": Object {},
        "type": "h1",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
